packages/ra-ui-materialui/src/input/BooleanInput.tsx

```diff
@@ -16,6 +16,7 @@ import { InputPropTypes } from './InputPropTypes';
 export const BooleanInput = (props: BooleanInputProps) => {
     const {
         className,
+        defaultValue = false,
```

Comment:
Well it's a boolean, a checkbox, which is unchecked by default, hence false

Reply:
If there is a default value, the form will set it and this has side effects. e.g. when a `post` with a `isPublished` set to `null` is edited, if the user changes the title, this will change the `isPublished` to `false`, which is unexpected. 

We didn't have it before, I wonder why we need it now.

Reply:
Because hook-form requires default values. This is a BooleanInput, not a NullableBooleanInput

Reply:
OK then.